src/test/java/com/google/edith/DealItemTest.java

```diff
@@ -0,0 +1,88 @@
+package com.google.edith;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+public class DealItemTest {
+
+  private DealItem dealItem;
+
+  @Before
+  public void setUp() {
+    dealItem = new DealItem();
+  }
+
+  @Test
+  public void dealItemInstantiates() {
+    Assert.assertTrue(dealItem instanceof DealItem);
+  }
+
+  @Test
+  public void canSetStore() {
+    dealItem.setStore("Walmart");
+    Assert.assertEquals("Walmart", dealItem.getStore());
+  }
+
+  @Test
+  public void priceParsedCorrectly() {
+    dealItem.setPrice("$25.5");
+    Assert.assertEquals(25.5, dealItem.getPrice(), .01);
+  }
+
+  @Test
+  public void badPriceDataHandled() {
+    dealItem.setPrice("");
+    Assert.assertEquals(0.0, dealItem.getPrice(), .01);
+  }
+
+  @Test
+  public void parseDoubleErrorHandledPrice() {
+    // Make sure unexpected data doesn't cause an error.
+    dealItem.setPrice("bad data");
+    Assert.assertEquals(0.0, dealItem.getPrice(), .01);
+  }
+
+  @Test
+  public void weightParsedCorrectly() {
+    dealItem.setWeight("15 fl oz");
+    Assert.assertEquals(15.0, dealItem.getWeight(), .01);
+  }
+
+  @Test
+  public void badWeightHandled() {
```

Comment:
Similar to how we name methods, it's best to name our test methods as descriptively as possible as well. This makes it easy to debug and isolate errors when there's a breakage or outage. E.g. If I run a presubmit test or check logs and see "negativeWeightReturns0" failed I know exactly what's wrong, whereas I might not understand what "bad" means on sight.

The rule of thumb is that test names should include the condition being tested and the expected outcome. 
E.g. 'multiply_twoNegativeValues_returnsPositiveValue' where multiply is the method, two negative values is the input, and positive value is the output.

Lmk if this makes and if so, let's update test names accordingly. Thanks!